python-checks/src/main/java/org/sonar/python/checks/PandasAddMergeParametersCheck.java

```diff
@@ -50,24 +88,39 @@ private static void verifyMergeCallParameters(SubscriptionContext ctx) {
     CallExpression callExpression = (CallExpression) ctx.syntaxNode();
     Optional.ofNullable(callExpression.calleeSymbol())
       .map(Symbol::fullyQualifiedName)
-      .filter(MERGE_METHODS::contains)
-      .ifPresent(fqn -> missingArguments(ctx, callExpression));
+      .filter(METHODS::contains)
+      .ifPresent(fqn -> missingArguments(fqn, ctx, callExpression));
   }
 
-  private static void missingArguments(SubscriptionContext ctx, CallExpression callExpression) {
+  private static void missingArguments(String fullyQualifiedName, SubscriptionContext ctx, CallExpression callExpression) {
     List<String> parameters = new ArrayList<>();
-    if (TreeUtils.argumentByKeyword("how", callExpression.arguments()) == null) {
-      parameters.add("how");
+    if (argumentIsMissing(fullyQualifiedName, Keywords.HOW, callExpression.arguments())) {
+      parameters.add(Keywords.HOW.getKeyword());
     }
-    if (TreeUtils.argumentByKeyword("on", callExpression.arguments()) == null) {
-      parameters.add("on");
+    if (argumentIsMissing(fullyQualifiedName, Keywords.ON, callExpression.arguments())) {
+      parameters.add(Keywords.ON.getKeyword());
     }
-    if (TreeUtils.argumentByKeyword("validate", callExpression.arguments()) == null) {
-      parameters.add("validate");
+    if (argumentIsMissing(fullyQualifiedName, Keywords.VALIDATE, callExpression.arguments())) {
+      parameters.add(Keywords.VALIDATE.getKeyword());
     }
     if (!parameters.isEmpty()) {
-      ctx.addIssue(callExpression, String.format(messages.get(parameters.size() - 1), parameters.toArray()));
+      parameters.add(fullyQualifiedName.substring(fullyQualifiedName.lastIndexOf('.') + 1));
```

Comment:
I  think we could make it a bit more explicit on what is happening here. We could refactor the String.format() into its own function where we could pass the arguments and name them properly for example:
```java
private static String generateMessage(String message, List<String> missingKeywords, String functionName){
    ...
}
```
This would help in understanding a bit the code. It is always a bit difficult to see the intent when using index on lists. This is why I would prefer having the arguments named so that if at some point the order of the MESSAGES came to change it would be easier to debug, knowing what `parameters.size() - 2` should be returning.

Reply:
I made the changes in 57fbeb2d6285498a551e7fbf581e7c47ab35da99.